pkg/cmd/cli/cmd/importimage.go

```diff
@@ -67,45 +68,147 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 		return err
 	}
 
+	insecure := cmdutil.GetFlagBool(cmd, "insecure")
 	from := cmdutil.GetFlagString(cmd, "from")
 	confirm := cmdutil.GetFlagBool(cmd, "confirm")
-	insecure := cmdutil.GetFlagBool(cmd, "insecure-repository")
+	all := cmdutil.GetFlagBool(cmd, "all")
+
+	if len(from) == 0 {
+		from = streamName
+	}
+
+	ref, err := imageapi.ParseDockerImageReference(streamName)
```

Comment:
There's only one ref, just like there's only one from, or one all, or one insecure.  Other refs later down can be more verbose.